File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    cnt0 = 0
    turn = 0
    
    while True:
        turn += 1
        cnt0 += s.count("0")
        s = s.replace("0", "")
        length = len(s)

        s = bin(int(length))
        if s == bin(1):
            answer = [turn, cnt0]
            break
        else:
            s = s[2:]
            

    return answer

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.swift
import Foundation

func solution(_ s:String) -> [Int] {
    var cur = s
    var pre = s
    var zeroCnt = 0
    var turn = 0
    
    while cur != "1" {
        cur = cur.replacingOccurrences(of: "0", with: "")
        zeroCnt += pre.count - cur.count
        
        // 이진법으로 변환
        cur = String(cur.count, radix: 2)
        turn += 1
        pre = cur
    }
    
    
    return [turn, zeroCnt]
}